keel-api/src/main/kotlin/com/netflix/spinnaker/config/EventsConfiguration.kt

```diff
@@ -0,0 +1,22 @@
+package com.netflix.spinnaker.config
+
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.event.SimpleApplicationEventMulticaster
+import org.springframework.core.task.TaskExecutor
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor
+
+@Configuration
+class EventsConfiguration {
+  @Bean
+  fun applicationEventTaskExecutor() =
+    ThreadPoolTaskExecutor().apply {
+      threadNamePrefix = "event-pool-"
+    }
+
+  @Bean
+  fun applicationEventMulticaster(applicationEventTaskExecutor: TaskExecutor) =
+    SimpleApplicationEventMulticaster().apply {
+      setTaskExecutor(applicationEventTaskExecutor)
+    }
```

Comment:
This is the magic required so that event listeners use a different thread. The way I've configured it there's an initial pool of 1 thread with an unbounded max size.